### PR TITLE
support typed msg sign for both string and object data

### PR DESCRIPTION
**Pull request by himanshuchawla009**

### CODE REVIEW

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts [hunk 1]
```diff
@@ -55,3 +62,54 @@ export async function fetchLegacyGasPriceEstimates(url: string): Promise<LegacyG
...
+          // for backward compatiblity we validate for both string and object type.
+          data = parameters.data;
+        }
+
+        assert.ok(data.primaryType in data.types, `Primary type of "${data.primaryType}" has no type definition.`);
+        const validation = jsonschema.validate(data, TYPED_MESSAGE_SCHEMA.properties);
+        assert.strictEqual(validation.errors.length, 0, "Signing data must conform to EIP-712 schema. See https://git.io/fNtcx.");
+        chainId = data.domain?.chainId;
+        if (chainId) {
+          assert.ok(!Number.isNaN(activeChainId), `Cannot sign messages for chainId "${chainId}", because Torus is switching networks.`);
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts:100]: just change torus to web3auth here

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/ethPrivatekeyUtils.ts [hunk 2]
```diff
@@ -78,19 +79,53 @@ export function getProviderHandlers({
...
+      if (!providerEngineProxy)
+        throw ethErrors.provider.custom({
+          message: "Provider is not initialized",
+          code: 4902,
+        });
+      const chainId = await providerEngineProxy.request<unknown, string>({ method: "eth_chainId" });
+      const finalChainId = Number.parseInt(chainId, isHexStrict(chainId) ? 16 : 10);
+      const params = {
+        ...msgParams,
+        version: "V1",
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-provider/src/providers/privateKeyProviders/ethPrivatekeyUtils.ts:92]: use enum SignTypedDataVersion.V1

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts [hunk 3]
```diff
@@ -55,3 +62,54 @@ export async function fetchLegacyGasPriceEstimates(url: string): Promise<LegacyG
...
+    assert.ok("from" in parameters, 'Params must include a "from" field.');
+    assert.ok(
+      typeof parameters.from === "string" && isValidAddress(parameters.from),
+      '"from" field must be a valid, lowercase, hexadecimal Ethereum address string.'
+    );
+    let data = null;
+    let chainId = null;
+    switch ((parameters as TypedMessageParams<unknown>).version) {
+      case "V1":
+        assert.ok(Array.isArray(parameters.data as unknown), "params.data must be an array.");
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts:79]: can this not be a string?

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/ethPrivatekeyUtils.ts (hunk 2 above)
Comment (himanshuchawla009) [reply packages/providers/ethereum-provider/src/providers/privateKeyProviders/ethPrivatekeyUtils.ts:92]: done

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts (hunk 3 above)
Comment (himanshuchawla009) [reply packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts:79]: done

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts (hunk 1 above)
Comment (himanshuchawla009) [reply packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/utils.ts:100]: done

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**